uber/config.py

```diff
@@ -1,124 +1,178 @@
+    on the state of the database).  See the comments in configspec.ini for explanations of the particilar
+    options, which are documented there.
+
+    This class has a single global instance called "c" which contains values of either type of config, e.g.
+    if you need to check whether dealer registration is open in your code, you'd say c.DEALER_REG_OPEN
     For all of the datetime config options, we also define BEFORE_ and AFTER_ properties, e.g. you can
-    check the booleans returned by state.BEFORE_PLACEHOLDER_DEADLINE or state.AFTER_PLACEHOLDER_DEADLINE
+    check the booleans returned by c.BEFORE_PLACEHOLDER_DEADLINE or c.AFTER_PLACEHOLDER_DEADLINE
     """
 
-    @property
-    def DEALER_REG_OPEN(self):
-        return self.AFTER_DEALER_REG_START and self.BEFORE_DEALER_REG_SHUTDOWN
-
-    @property
-    def BADGES_SOLD(self):
-        with Session() as session:
-            attendees = session.query(Attendee)
-            individuals = attendees.filter(or_(Attendee.paid == HAS_PAID, Attendee.paid == REFUNDED)).count()
-            group_badges = attendees.join(Attendee.group).filter(Attendee.paid == PAID_BY_GROUP,
-                                                                 Group.amount_paid > 0).count()
-            return individuals + group_badges
-
     def get_oneday_price(self, dt):
-        default = DEFAULT_SINGLE_DAY
-        return BADGE_PRICES['single_day'].get(dt.strftime('%A'), default)
+        default = self.DEFAULT_SINGLE_DAY
+        return self.BADGE_PRICES['single_day'].get(dt.strftime('%A'), default)
 
     def get_attendee_price(self, dt):
-        price = INITIAL_ATTENDEE
-        if PRICE_BUMPS_ENABLED:
-            for day, bumped_price in sorted(PRICE_BUMPS.items()):
+        price = self.INITIAL_ATTENDEE
+        if self.PRICE_BUMPS_ENABLED:
+            for day, bumped_price in sorted(self.PRICE_BUMPS.items()):
                 if (dt or datetime.now(UTC)) >= day:
                     price = bumped_price
         return price
 
     def get_group_price(self, dt):
-        return self.get_attendee_price(dt) - GROUP_DISCOUNT
+        return self.get_attendee_price(dt) - self.GROUP_DISCOUNT
+
+    @property
+    def DEALER_REG_OPEN(self):
+        return self.AFTER_DEALER_REG_START and self.BEFORE_DEALER_REG_SHUTDOWN
+
+    @property
+    def BADGES_SOLD(self):
+        with sa.Session() as session:
+            attendees = session.query(sa.Attendee)
+            individuals = attendees.filter(or_(sa.Attendee.paid == self.HAS_PAID, sa.Attendee.paid == self.REFUNDED)).count()
+            group_badges = attendees.join(sa.Attendee.group).filter(sa.Attendee.paid == self.PAID_BY_GROUP,
+                                                                    sa.Group.amount_paid > 0).count()
+            return individuals + group_badges
 
     @property
     def ONEDAY_BADGE_PRICE(self):
-        return self.get_oneday_price(localized_now())
+        return self.get_oneday_price(sa.localized_now())
```

Comment:
``localized_now()`` is a function which is defined in the ``uber.utils`` module, but that module isn't loaded until after this module (``uber.config``).  To get around the circular dependency, we use our ``sa`` module, which is just an alias for the top-level ``uber`` module.  So it's the same as what we're doing with our SQLAlchemy modules.